### ISSUE TITLE
No <cmd> advice on cfn-lint --help

### ISSUE THREAD

**Issue by martysweet** [martysweet/cfn-lint#110]
The help section should show some examples of how to use the command. Including what options for <cmd> there are.

```bash
user@a55ae7f0fda6:/# cfn-lint --help

  Usage: index [options] <cmd> <file>


  Options:

    -V, --version                        output the version number
    -p, --parameters <items>             List of params
    -p, --pseudo <items>                 List of pseudo overrides
    -G, --no-guess-parameters            Fail validation if a parameter with no Default is not passed
    -g, --only-guess-parameters <items>  Guess the provided parameters, and fail validation if a parameter with no Default is passed
    -h, --help                           output usage information
```

**Comment by RazzM13** [comment]
In PR #136, I've updated the `--help` output to the following:
```
$ cfn-lint --help

  Usage: cfn-lint [options] [command]

  Options:

    -V, --version       output the version number
    -h, --help          output usage information

  Commands:

    validate [options]
    docs

no command given!
```
Also, I've taken the liberty of showing the usage prompt in cases where either the command or the argument is missing.

**Comment by RazzM13** [comment]
PS: perhaps the `no command given!` and `missing required argument` messages should have the first letter uppercased and perhaps an exclamation mark could be added to the latter.

**Comment by martysweet** [comment]
@RazzM13, does this allow people to run `cfn-lint validate --help` to see the related options?

**Comment by RazzM13** [comment]
Hi @martysweet, yes it does for `validate` as well as for `docs`.

```
$ cfn-lint validate --help

  Usage: validate [options]

  Options:

    -p, --parameters <items>             List of params
    -p, --pseudo <items>                 List of pseudo overrides
    --guess-parameters                   Guess any parameters that are not explicitely passed in and have no Default. This is the default behaviour.
    -G, --no-guess-parameters            Fail validation if a parameter with no Default is not passed
    -g, --only-guess-parameters <items>  Guess the provided parameters, and fail validation if a parameter with no Default is passed
    -v, --verbose                        Verbose error messages
    -h, --help                           output usage information

 missing required argument
```

```
$ cfn-lint docs --help

  Usage: docs [options]

  Options:

    -h, --help  output usage information

 missing required argument
```

And even when no arguments have been provided:
```
$ cfn-lint validate

  Usage: validate [options]

  Options:

    -p, --parameters <items>             List of params
    -p, --pseudo <items>                 List of pseudo overrides
    --guess-parameters                   Guess any parameters that are not explicitely passed in and have no Default. This is the default behaviour.
    -G, --no-guess-parameters            Fail validation if a parameter with no Default is not passed
    -g, --only-guess-parameters <items>  Guess the provided parameters, and fail validation if a parameter with no Default is passed
    -v, --verbose                        Verbose error messages
    -h, --help                           output usage information

 missing required argument
```
